### PR TITLE
JAMES-3786 Mailbox index could support dedicated language

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+					"tokenizer": "uax_url_email",
+					"filter": ["lowercase", "stop", "english_snowball"]
+				}
+			},
+			"tokenizer": {},
+			"filter": {
+				"english_snowball": {
+					"type": "snowball",
+					"language": "English"
+				},
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json:33]: As stated by @quantranhong1999 before it seems english_snowball is declared in the analyzer snowball_keep_mail_and_token that is not used anywhere...

I think you should clean up `english_snowball` filter and `snowball_keep_mail_and_token` analyzer though?

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+				"french_elision": {
+					"type": "elision",
+					"articles_case": true,
+					"articles": [
+						"l", "m", "t", "qu", "n", "s",
+						"j", "d", "c", "jusqu", "quoiqu",
+						"lorsqu", "puisqu"
+					]
+				},
```
[the remark this answers is not in the document]
Comment (quantranhong1999) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json]: This is from the [default `french` analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html#french-analyzer) of ES. Not sure why you want no predefined value here. IMO we can just put some default values here and let people modify it further.

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (quantranhong1999) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json]: cleaned up

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/example_french_index.json]: Ok, that's just that it feels rather arbitrary...

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**